File: 567-permutation-in-string/permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        if(s2.contains(s1)) return true;
        char ak[] = s1.toCharArray();
        Arrays.sort(ak);
        s1 = new String(ak);
        for(int i = 0; i<=s2.length() - s1.length(); i++){
            String m = s2.substring(i, i + s1.length());
            char kb[] = m.toCharArray();
            Arrays.sort(kb);
            m = new String(kb);
            if(m.equals(s1)) return true;
        }
        return false;
    }
}

File: 567-permutation-in-string/permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if(s1.size() > s2.size()) return false;
    
        // if(s1.size()==1 && s2.find(s1) != string :: npos) return true;
        sort(s1.begin(), s1.end());
        for(int i = 0; i<=s2.size() - s1.size(); i++){
            string ak = s2.substr(i, s1.size());
            sort(ak.begin(), ak.end());
            if(ak == s1) return true;
        }
        return false;
    }
};